web/modules/custom/dpl_loans/src/Plugin/Block/LoanListBlock.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+namespace Drupal\dpl_loans\Plugin\Block;
+
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\dpl_react_apps\Controller\DplReactAppsController;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides user loans list.
+ *
+ * @Block(
+ *   id = "dpl_loans_list_block",
+ *   admin_label = "List user loans"
+ * )
+ */
+class LoanListBlock extends BlockBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * Drupal config factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  private ConfigFactoryInterface $configFactory;
+
+  /**
+   * LoanListBlock constructor.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $configFactory
+   *   Drupal config factory to get FBS and Publizon settings.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ConfigFactoryInterface $configFactory) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->configuration = $configuration;
+    $this->configFactory = $configFactory;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('config.factory'),
+    );
+  }
+
+  /**
+   * Gets threshold config.
+   *
+   * @return string
+   *   Returns the threshold config.
+   */
+  public function getThresholdConfig(): string {
+    $generalSettings = $this->configFactory->get('dpl_library_agency.general_settings');
+    return $generalSettings->get('threshold_config');
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @return mixed[]
+   *   The app render array.
+   */
+  public function build() {
+    $loanListSettings = $this->configFactory->get('dpl_loan_list.settings');
+    $context = ['context' => 'Loan list'];
+    $contextAria = ['context' => 'Loan list (Aria)'];
```

Comment:
Great with multiple contexts! I think we have to few in the material app (namely one).

One thing that I would like to mention is that, with the translation server task we are looking into using a tool for scanning the codebase for translation functions. It is probably using some kind of build in reflection functionality in PHP. But one thing I have noticed is that:
* You have to write literal strings for the source string (the first argument for `t()`). That's not something new, but I have seen places in our common codebase where `t()` is being used when errors are thrown.
*Context should be written literally everytime a translation function is being used. So it is not possible to reuse `$context` as it is being done here.

This is just to mention it. And I guess there is quite some refactoring to be done if we stick with the current idea of how we should scan the code. Alternatively we could put the whole data prop array (at least the text part) in .inc files to clean it up a bit.

Reply:
Why scan the code base... will Drupal not build an database table over all translatable strings that could be "dump" and used. Repeating the context string (in some cases 100 of times in the code base) will only lead to C&P errors and typos.

I think it would be much better idea to reuse the Drupal parts that already collect translations from the code base.

Reply:
It is because we want to be able to register all strings without rendering the pages/apps.
It is the same principle that the translation server of drupal.org, https://localize.drupal.org, is using.